src/BotPlutusInterface/CardanoCLI.hs

```diff
@@ -349,11 +272,14 @@ submitTx pabConf tx =
       )
       (const ())
 
-txInOpts :: PABConfig -> TxInfo -> Set TxIn -> ([Text], ExBudget)
-txInOpts pabConf txInfo =
+txInOpts :: SpendBudgets -> PABConfig -> Set TxIn -> ([Text], ExBudget)
+txInOpts spendIndex pabConf =
   foldMap
     ( \(TxIn txOutRef txInType) ->
-        let (opts, exBudget) = scriptInputs txOutRef txInType
+        let (opts, exBudget) =
+              scriptInputs
+                txInType
+                (Map.findWithDefault mempty txOutRef spendIndex)
```

Comment:
This part feels a bit unsafe, as at the point when budgets are estimated and non-draft transaction body being built, missing `txOutRef` key in `spendIndex` means that something went wrong. It should not happen, but still.

Maybe return type of `txInOpts` should be `Either ... ([Text], ExBudget)` instead of `([Text], ExBudget)`. But I was not sure if I  should spend more time to refine this, as it will require more complex logic to separate `buildTx` and `buildDraftTx`. 
And if I'm not mistaken, worst thing that could happen now when `ExBudget` will be missing in `buildTx` for input or minting policy , is that transaction will fail during submission with "budget overspent" error.

Reply:
This should be okay